ts/features/idpay/details/components/TimelineOperationListItem.tsx

```diff
@@ -2,218 +2,393 @@ import {
+  <ListItemTransaction
+    paymentLogoIcon={
+      <Icon name={"institution"} color="grey-300" testID="ibanLogoTestID" />
     }
-  };
+    title={I18n.t(
+      `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.ADD_IBAN`
+    )}
+    subtitle={getOperationSubtitle(props.operation.operationDate)}
+    transactionStatus={"success"}
+    transactionAmount=""
+    onPress={props.onPress}
+    testID={props.testID}
+  />
+);
+
+const OnboardingOperationListItem = (
+  props: ListItemProps<OnboardingOperationDTO>
+) => (
+  <ListItemTransaction
+    paymentLogoIcon={
+      <Icon name={"checkTick"} color="grey-300" testID="onboardingLogoTestID" />
+    }
+    title={I18n.t(
+      `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.ONBOARDING`
+    )}
+    subtitle={getOperationSubtitle(props.operation.operationDate)}
+    transactionStatus={"success"}
+    transactionAmount={""}
+    testID={props.testID}
+  />
+);
+
+const RefundOperationListItem = (props: ListItemProps<RefundOperationDTO>) => {
+  const { operationDate, operationType, amount } = props.operation;
+  const isRejected = operationType === RefundOperationTypeEnum.REJECTED_REFUND;
+
+  const operationLogo = (
+    <Icon name={"refund"} color="grey-300" testID="refundLogoTestID" />
+  );
+  const title = I18n.t(
+    `idpay.initiative.details.initiativeDetailsScreen.configured.operationsList.operationDescriptions.REFUND`
+  );
+  const subtitle = getOperationSubtitle(operationDate);
 
-  const operationStatus = getOperationStatus();
+  if (isRejected) {
+    return (
+      <ListItemTransaction
+        title={title}
+        subtitle={subtitle}
+        paymentLogoIcon={operationLogo}
+        transactionStatus={"failure"}
+        badgeText={getBadgeTextByTransactionStatus("failure")}
+        onPress={props.onPress}
+        testID={props.testID}
+      />
+    );
+  }
 
   return (
     <ListItemTransaction
-      title={getOperationTitle()}
-      subtitle={`${localeDateFormat(
-        operation.operationDate,
-        I18n.t("global.dateFormats.fullFormatShortMonthLiteral")
-      )}, ${getHourAndMinuteFromDate(
-        operation.operationDate
-      )}${getOperationAmount()}`}
-      paymentLogoIcon={getOperationLogo(operation)}
-      transactionStatus={operationStatus}
-      badgeText={getBadgeTextByTransactionStatus(
-        operationStatus as ListItemTransactionStatusWithBadge
-      )}
-      transactionAmount={getTransactionAmountLabel(operation)}
-      onPress={onPress}
+      title={title}
+      subtitle={subtitle}
+      paymentLogoIcon={operationLogo}
+      transactionStatus={"success"}
+      transactionAmount={`${formatAbsNumberAmountOrDefault(amount)} €`}
+      onPress={props.onPress}
+      testID={props.testID}
     />
   );
 };
 
-const TimelineOperationListItemSkeleton = () => (
+const SuspendOperationListItem = (
```

Comment:
I believe that dividing all the listItems like this, rather than just picking the props with the original switch and then using them to render, adds a lot of verbosity for more downsides than benefits; what do you think?

Reply:
Could you give me some examples? 
You mean, instead of React components we should just make regular function to return the props to pass to a `ListItemTransaction`? Wouldn't you find it difficult to debug and/or mantain instead of just regular React components?

Reply:
I tried your suggestion https://github.com/pagopa/io-app/pull/5200/commits/157885224a8634a1773c9d02b52748a382fede16
I agree with you, I think it's more readable. Let me know!